src/main/java/org/jvnet/hudson/plugins/shelveproject/ShelveProjectAction.java

```diff
@@ -27,11 +27,11 @@ public ShelveProjectAction(Item item) {
 
   @Override
   public String getIconFileName() {
-    return getShelveIconPath();
+    return getShelveIconPath(getItem());
   }
 
-  private static String getShelveIconPath() {
-    return Jenkins.getInstance().hasPermission(SHELVE_PERMISSION) ? ACTION_ICON_PATH : null;
+  private static String getShelveIconPath(Item item) {
+    return (item != null) ? item.hasPermission(SHELVE_PERMISSION) ? ACTION_ICON_PATH : null : null;
```

Comment:
The method should not be static anymore. You can also simplify the code, instead of two ternary operators, hard to read, you can use one: `item != null &&  item.hasPermission(SHELVE_PERMISSION) ? ...`